### ISSUE TITLE
Migration from 2.16 to 2.17. How to use drmSessionManagerProvider instead setDrmSessionManager

### ISSUE THREAD

**Issue by adamheinrich-oke** [google/ExoPlayer#10208]
In my dashfactory I was calling:
 
```
                DashMediaSource.Factory(
                        DefaultDashChunkSource.Factory(mediaDataSourceFactory),
                        buildDataSourceFactory(context, false, bandwidthMeter, userAgent)
                )       .setDrmSessionManager(customManager)
                        .createMediaSource(getMediaItem(streamUrl, MIME_TYPE_DASH))
```

However after the update to 2.17.1 I have no longer access to `.setDrmSessionManager()` 
I was passing customManager created in other function in a such way:
```
fun (){....
        return DefaultDrmSessionManager.Builder()
                .setUuidAndExoMediaDrmProvider(uuid, AppManagedProvider(mediaDrm))
                .setMultiSession(true)
                .build(drmCallback)
           
                }
```
                
 So I created DefaultDrmSessionManager with Bulder() call.
However now I cannot use this object as drmSessionManagerProvider  is needed. 
I got type mismatch. 
Required: DrmSessionManagerProvider?
Found: DefaultDrmSessionManager

What is the proper way to use DrmSessionManagerProvider? I see I can use `DefaultDrmSessionManagerProvider()`
but I'd like to change i'ts properties as I did with previous manager via Builder() call

I have not come across any migration guide. Thanks.

**Comment by icbaker** [comment]
tl;dr You should be able to do this by creating a very simple `DrmSessionManagerProvider` that just always returns the same instance, e.g. with a lambda:

```java
.setDrmSessionManagerProvider({ unusedMediaItem -> customManager })
```

This is covered here: https://exoplayer.dev/drm.html#using-a-custom-drmsessionmanager

---

Longer version:

This method (and some other DRM related methods) were [deprecated on the `MediaSourceFactory` interface in 2.13.0](https://github.com/google/ExoPlayer/blob/r2.13.0/library/core/src/main/java/com/google/android/exoplayer2/source/MediaSourceFactory.java#L92), including some `@deprecated` javadoc that tried to assist in migrating. Unfortunately the `@Deprecated` annotation was not propagated to the `DashMediaSource.Factory` implementation, even up to [version 2.16.0](https://github.com/google/ExoPlayer/blob/r2.16.0/library/dash/src/main/java/com/google/android/exoplayer2/source/dash/DashMediaSource.java#L182). This means that when calling the method on something with compile-time type of `DashMediaSource.Factory` you would not have seen a deprecation warning, or the migration instructions, in Android Studio, Gradle etc. 

Sorry about that oversight!

**Comment by adamheinrich-oke** [comment]
it works thank you @icbaker
